crates/holochain/src/conductor/conductor/tests/agent_key_revocation.rs

```diff
@@ -0,0 +1,219 @@
+            ],
+        )
+        .await
+        .unwrap();
+    let agent_key = app.agent().clone();
+    let zome_1 = SweetZome::new(
+        app.cells()[0].cell_id().clone(),
+        coordinator_zomes_1[0].name.clone(),
+    );
+    let zome_2 = SweetZome::new(
+        app.cells()[1].cell_id().clone(),
+        coordinator_zomes_2[0].name.clone(),
+    );
+    let create_fn_name = "create_entry";
+    let read_fn_name = "get_post";
+
+    // no agent key provided, so DPKI should be installed
+    // and the generated agent key be valid
+    let dpki = conductor
+        .running_services()
+        .dpki
+        .expect("dpki must be running");
+    let key_state = dpki
+        .state()
+        .await
+        .key_state(agent_key.clone(), Timestamp::now())
+        .await
+        .unwrap();
+    assert_matches!(key_state, KeyState::Valid(_));
+
+    // deleting a non-existing key should fail
+    let non_existing_key = AgentPubKey::from_raw_32(vec![0; 32]);
+    let result = conductor
+        .clone()
+        .revoke_agent_key_for_app(non_existing_key.clone(), app.installed_app_id().clone())
+        .await;
+    assert_matches!(
+        result,
+        Err(ConductorError::DpkiError(DpkiServiceError::DpkiAgentMissing(key))) if key == non_existing_key
+    );
+
+    // writing to the cell should succeed
+    let action_hash_1: ActionHash = conductor.call(&zome_1, create_fn_name, ()).await;
+    let action_hash_2: ActionHash = conductor.call(&zome_2, create_fn_name, ()).await;
+
+    // deleting the key should succeed
+    let result = conductor
+        .clone()
+        .revoke_agent_key_for_app(agent_key.clone(), app.installed_app_id().clone())
+        .await;
+    assert_matches!(result, Ok((_, _)));
+
+    // key should be invalid
+    let key_state = dpki
+        .state()
+        .await
+        .key_state(agent_key.clone(), Timestamp::now())
+        .await
+        .unwrap();
+    assert_matches!(key_state, KeyState::Invalid(_));
+
+    // deleting agent key again should return a "key invalid" error
+    let result = conductor
+        .clone()
+        .revoke_agent_key_for_app(agent_key.clone(), app.installed_app_id().clone())
+        .await;
+    assert_matches!(result, Err(ConductorError::DpkiError(DpkiServiceError::DpkiAgentInvalid(invalid_key, _))) if invalid_key == agent_key);
+
+    // reading an entry should still succeed
+    let result: Option<Record> = conductor.call(&zome_1, read_fn_name, action_hash_1).await;
+    assert!(result.is_some());
+    let result: Option<Record> = conductor.call(&zome_2, read_fn_name, action_hash_2).await;
+    assert!(result.is_some());
+
+    // creating an entry should fail now for both cells
+    let result = conductor
+        .call_fallible::<_, ActionHash>(&zome_1, create_fn_name, ())
+        .await;
+    if let Err(ConductorApiError::CellError(CellError::WorkflowError(workflow_error))) = result {
+        assert_matches!(
+            *workflow_error,
```

Comment:
This is a `Box<Error>` - is there an easier way of matching against the innermost error?

Reply:
no :(